src/careamics/prediction_utils/lvae_prediction.py

```diff
@@ -0,0 +1,92 @@
+"""Module containing pytorch implementations for obtaining predictions from an LVAE."""
+
+from typing import Optional
+
+import torch
+
+from careamics.models import LVAE
+
+# TODO: convert these functions to lightning module `predict_step`
+#   -> mmse_count will have to be an instance attribute?
+
+
+def lvae_predict_single_sample(
+    model: LVAE, input: torch.Tensor
+) -> tuple[torch.Tensor, Optional[torch.Tensor]]:
+    """
+    Generate a single sample prediction from an LVAE model, for a given input.
+
+    Parameters
+    ----------
+    model : LVAE
+        Trained LVAE model.
+    input : torch.tensor
+        Input to generate prediction for. Expected shape is (S, C, Y, X).
+
+    Returns
+    -------
+    tuple of (torch.tensor, optional torch.tensor)
+        The first element is the sample prediction, and the second element is the
+        log-variance. The log-variance will be None if `model.predict_logvar is None`.
+    """
+    model.eval()  # Not in original predict code: effects batch_norm and dropout layers
+    with torch.no_grad():
+        output: torch.Tensor
+        output, _ = model(input)  # 2nd item is top-down data dict
+
+    # TODO: what if this is likelihood_NM (Not implemented at the moment)
+    # not sure how this will be fully implemented
+    #   -> potentially log_var is always None ?
+    likelihood_obj = model.likelihood_gm
+
+    # presently, get_mean_lv just splits the output in 2 if predict_logvar=True,
+    #   optionally clips the logvavr if logvar_lowerbound is not None
+    sample_prediction, log_var = likelihood_obj.get_mean_lv(output)
```

Comment:
Well, it is debatable. I would not expose how the likelihood object operates in a standalone prediction module. I would let the likelihood object to decide on how to return the log_var.
I see  that your approach is less code, but the one present is more modular.

Reply:
I agree that using the likelihood object is potentially misleading and misrepresents the intention of the code since we are not using the true functionality of the likelihood. And I did actually start off with the implementation above (with also the optional clipping of the lowerbound of the logvar). However, I reverted back to using the `get_mean_lv` method because I still had to use two attributes from the likelihood object, namely `predict_logvar` and `logvar_lowerbound`, and I wanted to keep consistency with the training and validation steps (although I think the mean & logvar splitting happens within `reconstruction_loss`).

Additionally I think we could consider changing the output of the model itself so that the `mean` and `logvar` are already split apart. i.e. `mean, logvar, topdown_dict = model(x)`.

Therefore I reckon we make a note to refactor this when we have more of the pipeline in place, because the choice we make now might not be optimal for the final product.

Reply:
> well, at this point, anything and everything which keeps the code working is perfectly fine for me. :P

Yes I didn't want to introduce too many drastic changes, especially on sections I don't fully understand, which may lead to unexpected behaviour down the road. We will leave this to be addressed at a later date.

Reply:
Having said that, since #205, the likelihood is no longer an attribute of the LVAE model. To solve this issue I think I will simply pass the likelihood object as an argument to the prediction functions. If I want to remove dependence on the likelihood model as @federico-carrara suggests then I will have to pass both `predict_logvar` and `logvar_lowerbound` to the prediction functions.